planner_server:
  ros__parameters:
    planner_plugins: ["GridBased"]
    expected_planner_frequency: 10.0
    GridBased:
      plugin: "nav2_navfn_planner/NavfnPlanner"

global_costmap:
  ros__parameters:
    global_frame: map
    robot_base_frame: base_link
    update_frequency: 1.0
    publish_frequency: 1.0
    width: 10.0
    height: 10.0
    resolution: 0.05
    rolling_window: false
    always_send_full_costmap: true
    plugins: ["static_layer", "obstacle_layer", "inflation_layer"]

    static_layer:
      plugin: "nav2_costmap_2d::StaticLayer"

    obstacle_layer:
      plugin: "nav2_costmap_2d::ObstacleLayer"
      observation_sources: scan
      scan:
        topic: scan
        sensor_frame: laser_frame
        data_type: LaserScan
        clearing: true
        marking: true

    inflation_layer:
      plugin: "nav2_costmap_2d::InflationLayer"
      inflation_radius: 0.1
